autoload/searchx.vim

```diff
@@ -356,3 +357,6 @@ function! s:find_matches(input, curpos) abort
   return { 'matches': l:matches, 'current': l:current }
 endfunction
 
+function! searchx#set_history(pattern, dir) abort
+  execute 'normal!' (a:dir == 1 ? '/' : '?') . a:pattern
```

Comment:
I'm warried to this. I think `normal!` has side-effect like cursor jumping, hlsearch activation etc.
I'll try this patch.

Reply:
In the neovim environment at hand, the behavior was just to add it to the search history without any side effects, but there may be some settings or plugins that are affecting the behavior. Please check.

Reply:
I can't narrow `/` history with `<C-p>` or `<Up>`.
But you're right. The latest master is broken too.